### DISCUSSION TITLE
EADDRINUSE and ENOENT errors when trying to sync with GitHub

### DISCUSSION THREAD

**Discussion by phizeroth** [requarks/wiki#3000] [Help / Questions]
My friend and i share a small server and each of us have our own Linux wiki-js set up running on PM2.  All works fine except he can't sync with GitHub, but i've had no issues.  When he tries to connect with Git he gets these two errors in the logs:

```
1|bubbawik | Error: listen EADDRINUSE: address already in use 0.0.0.0:3443
1|bubbawik |     at Server.setupListenHandle [as _listen2] (net.js:1313:16)
1|bubbawik |     at listenInCluster (net.js:1361:12)
1|bubbawik |     at doListen (net.js:1498:7)
1|bubbawik |     at args.<computed> (/usr/lib/node_modules/pm2/node_modules/event-loop-inspector/index.js:138:29)
1|bubbawik |     at processTicksAndRejections (internal/process/task_queues.js:85:21)

1|bubbawik | Error: Error: spawn   ENOENT
1|bubbawik |     at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
1|bubbawik |     at onErrorNT (internal/child_process.js:469:16)
1|bubbawik |     at args.<computed> (/usr/lib/node_modules/pm2/node_modules/event-loop-inspector/index.js:138:29)
1|bubbawik |     at processTicksAndRejections (internal/process/task_queues.js:84:21)
1|bubbawik |     at GitExecutorChain.onFatalException (/var/wiki/bubba/node_modules/simple-git/src/lib/runners/git-executor-chain.js:60:87)
1|bubbawik |     at GitExecutorChain.<anonymous> (/var/wiki/bubba/node_modules/simple-git/src/lib/runners/git-executor-chain.js:51:28)
1|bubbawik |     at Generator.throw (<anonymous>)
1|bubbawik |     at rejected (/var/wiki/bubba/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)
```

I've tried changing the SSL port in config.yml from 3443 to 4443 to see if there was a conflict with my own instance, with no effect and the EADDRINUSE error still referred to port 3443.  Not sure what else to try, any guidance would be appreciated.

**Comment by NGPixel** [comment]
Seems like you have 2 or more processes running at the same time. Check your pm2 config to make sure only 1 instance is running.

**Reply by phizeroth** [reply]
I run both his and my wiki in pm2.  I shut down both, rebooted and started only his, and it seems to have eliminated the EADDRINUSE error but he's still getting the ENOENT error

**Reply by NGPixel** [reply]
Looks like your working directory set in pm2 is wrong. Make sure to run from / and not /server.
